api/src/test/java/by/borge/jarl/JarlTests.java

```diff
@@ -0,0 +1,22 @@
+package by.borge.jarl;
+
+import org.junit.jupiter.api.Test;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.Map;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class JarlTests {
+    @Test
+    void simplePlan() throws IOException {
+        var file = new File("../core/src/test/resources/rego/simple/plan.json");
```

Comment:
What version of Java is required for this? Do we make a difference here between core classes and tests?

Reply:
The `var` keyword was introduced in 10, I think. We also use `record`, which is new since 14.
If we drop that "sugar", we're probably Java 7-8 compatible.
We could probably scale back to Java 8 compliancy (extended support until 2030), given that most production code is likely to keep being confined to Clojure ..
The question is, how much adoption will we miss out on if we don't want to be yet another lib stuck in the past?

Reply:
I'm fine with modern Java being required, and I definitely don't see us supporting anything below Java 11. Since this is a farily small component, I guess we can just provide a Java 11 version should it be required later on.

Reply:
Agreed! Let's see how adoption shakes out first.